src/polytope/utils/uuidgen.py

```diff
@@ -0,0 +1,111 @@
+import random
+from typing import List
+
+# collision probability in 150,000 entries ~ 1%
+# lowercase alphabet + digit except [l, 1, o, 0]
+DEFAULT_ALPHABET = "abcdefghijkmnpqrstuvwxyz23456789"
+DEFAULT_LENGTH = 8
+
+
+class PolytopeUUID:
+    """UUID generator class."""
+
+    def __init__(
+        self, alphabet: str = DEFAULT_ALPHABET, length: int = DEFAULT_LENGTH
+    ):
+        """! PolytopeUUID class initializer.
+
+        @param alphabet     alphabet for generating uuid
+        @param length       length of uuid
+        """
+        if len(alphabet) == 0:
+            raise ValueError("alphabet must be a non-empty string.")
+        if len(set(alphabet)) != len(alphabet):
+            raise ValueError("alphabet must consist of distinct characters.")
+        if length <= 0:
+            raise ValueError("length must be positive.")
+
+        self._alphabet = alphabet
+        self._length = length
+
+    @property
+    def alphabet(self):
+        """! An alphabet property for generating uuid."""
+        return self._alphabet
+
+    @property
+    def length(self):
+        """! A length property of uuid."""
+        return self._length
+
+    @alphabet.setter
+    def alphabet(self, value: str):
+        """! A getter method for alphabet property."""
+        if len(value) == 0:
+            raise ValueError("alphabet must be a non-empty string.")
+        if len(set(value)) != len(value):
+            raise ValueError("alphabet must consist of distinct characters.")
+
+        self._alphabet = value
+
+    @length.setter
+    def length(self, value: int):
+        """! A getter method for length property."""
+        if value <= 0:
+            raise ValueError("length must be positive.")
+
+        self._length = value
+
+    def uuid(self) -> str:
+        """! A method for generating uuid."""
+        char_list = [random.choice(self.alphabet) for _ in range(self.length)]
+        return "".join(char_list)
+
+    def uuid_bulk(self, count: int) -> List[str]:
+        """! A method for bulk generating uuid.
+
+        @param count    number of uuid to generate
+        """
+        total = len(self.alphabet) ** self.length
+
+        if count <= 0:
+            raise ValueError("count must be positive.")
+
+        if count > total:
+            raise ValueError("count is too large to generate distinct uuids")
```

Comment:
(question) `total` seems to be too large. Also, if `count` is very close to `total`, then this method will take a very long time.
Is there a clever way to check if `count` is **reasonably small** compared to `total`?

Reply:
How about comparing `log(count)` with `self.length * log(len(self.alphabet))`?

Reply:
It seems like a good idea to seperate the uuid generation method into two.
* If `total` is small enough to manage as `int` type, then do as it is now.
* If `total` is too large, then set an appropriate upper bound on `count` and do a classical randomized method.

Reply:
I implemented `__uuid_bulk_large` method, which called when `total` doesn't fit `int` type.

https://github.com/Studio-Polytope/Polytope/blob/3ab4c864db2017a08702b4d82dbc11dd9a029314/src/polytope/utils/uuidgen.py#L65